reframe/core/containers.py

```diff
@@ -5,15 +5,20 @@
 
 import abc
 
-import reframe.core.fields as fields
 import reframe.utility as util
 import reframe.utility.typecheck as typ
+from reframe.core.meta import RegressionTestMeta
 
 
-_STAGEDIR_MOUNT = '/rfm_workdir'
+class _ContainerMeta(typ.ConvertibleType, RegressionTestMeta, abc.ABCMeta):
+    '''Metaclass for container platforms.
+
+    This essentially combines the ABCMeta with our standard metaclass that
+    provides the variables.
+    '''
 
 
-class ContainerPlatform(abc.ABC):
+class ContainerPlatform(abc.ABC, metaclass=_ContainerMeta):
```

Comment:
I just thinking out loud... Whenever, I see the use if `RegressionTestMeta` I wonder wether it should be implemented using a `RegrssionMixin`...

Reply:
Maybe you are right, I'll try that out. The intent was to bring in the "variable" definitions.